GCRCatalogs/register.py

```diff
@@ -558,12 +590,22 @@ def get_root_dir():
     return _config_register.root_dir
 
 
-def set_root_dir(path):
+def set_root_dir(path, write_to_config=False):
     """
     Sets runtime root_dir to *path*.
+
+    Returns
+    -------
+    If write_to_config, return old persisted value (which may be None);
+    else None.
     """
     _config_register.root_dir = path
 
+    if write_to_config:
+        return _config_register.persist_root_dir()
+    else:
+        return None
```

Comment:
A function that does not encounter any return statement will automatically returns None

```suggestion
```